3_javascript/javascript1/week3/script.js

```diff
@@ -0,0 +1,217 @@
+
+function logOutNotesFormatted() {
+    for (let i = 0; i < notes.length; i++) {
+        console.log(`The note with id: ${notes[i].id}, has the following note text: ${notes[i].content} `);
+    }
+};
+
+logOutNotesFormatted(); // should log out the text below
+// The note with id: 1, has the following note text: Pick up groceries
+// The note with id: 2, has the following note text: Do laundry
+
+//--------- Unique feature --------------------------------
+
+//What can it be fx.? :)
+
+
+
+
+
+
+
+// -----------CactusIO-interactive (Smart phone usage app) optional -------------------
+
+//PART1: Adding an activity ______________________________________
+
+const activities = [];  // [{}]  or [{}] would also work?
+
+function addActivity(date, activity, duration) {  //date: string, activity: string; duration: minutes
+
+    let activitiesDataDuration = { activity: activity, date: date, duration: duration };
+    activities.push(activitiesDataDuration);
+    return activities;
+}
+
+let gottenActivity;
+let gottenDate;
+let gottenDuration;
+
+gottenDate = prompt('add a date fx. 23/7-2023', '23/7-2023');
+gottenActivity = prompt('add an activity fx. Youtube', 'Youtube');
+gottenDuration = prompt('Enter a number as activity duration hours fx.', 30);
+addActivity(gottenDate, gottenActivity, gottenDuration);
+
+gottenDate = prompt('add a date fx. 23/7-2023', '23/7-2023');
+gottenActivity = prompt('add an activity fx. Instagram', 'Instagram');
+gottenDuration = prompt('Enter a number as activity duration hours fx.', 40);
+addActivity(gottenDate, gottenActivity, gottenDuration);
+
+gottenDate = prompt('add a date fx. 23/7-2023', '23/7-2023');
+gottenActivity = prompt('add an activity fx. Facebook', 'Facebook');
+gottenDuration = prompt('Enter a number as activity duration hours fx.', 15);
+addActivity(gottenDate, gottenActivity, gottenDuration);
+
+console.log(activities);
+
+// PART2: Adding an activity ____________________________________
+
+function showStatus(activities) {
+
+    if (activities.length === 0) {
+        let NoactivitiesAdded = console.log(`You have not added any activities. 
+        Add some activities before to request seeing the status of activities`);
+        //Should I call the function again???
+        return NoactivitiesAdded;
+
+    } else {
+        console.log(`You have added ${activities.length} activities`);
+        for (let i = 0; i < activities.length; i++) {
+            console.log(`Activities ${i + 1}: ${activities[i].activity}`);
+        }
+
+        let totalSmartphoneUsagedurationInMinuits = 0;
+        for (let j = 0; j < activities.length; j++) {
+            totalSmartphoneUsagedurationInMinuits = totalSmartphoneUsagedurationInMinuits + activities[j].duration;
+            //Why it gives a different digit whenever I run the codes???
+        }
+        console.log(`Activities amount to ${totalSmartphoneUsagedurationInMinuits} min. of usage on ${activities[0].date}`);
```

Comment:
If I understand correctly the activities can have a different date each, if so, using `${activities[0].date}` can give you an incorrect result. I recommend to remove the whole `of usage on date` part since its not part of the assignment anyway :)

Reply:
Thank you :)
The dates are the same, as we should find the usage duration for the predefined date. I have changed ${activities[0].date} it to ${activities.date} and now it works correctly.